desktop/core/src/desktop/js/utils/hooks/UseFileUpload/useChunkUpload.ts

```diff
@@ -0,0 +1,151 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useEffect, useState } from 'react';
+import useSaveData from '../useSaveData';
+import useQueueProcessor from '../useQueueProcessor';
+import {
+  DEFAULT_CHUNK_SIZE,
+  DEFAULT_CONCURRENT_CHUNK_UPLOAD,
+  FileUploadStatus
+} from '../../constants/storageBrowser';
+import useLoadData from '../useLoadData';
+import { TaskServerResponse, TaskStatus } from '../../../reactComponents/TaskBrowser/TaskBrowser';
+import {
+  createChunks,
+  getChunksCompletePayload,
+  getChunkItemPayload,
+  getChunkSinglePayload,
+  getStatusHashMap,
+  getTotalChunk,
+  UploadChunkItem,
+  UploadItem
+} from './util';
+
+interface UseUploadQueueResponse {
+  addFiles: (item: UploadItem[]) => void;
+  removeFile: (item: UploadItem) => void;
+  isLoading: boolean;
+}
+
+interface ChunkUploadOptions {
+  onStatusUpdate: (item: UploadItem, newStatus: FileUploadStatus) => void;
+  onComplete: () => void;
+}
+
+const useChunkUpload = ({
+  onStatusUpdate,
+  onComplete
+}: ChunkUploadOptions): UseUploadQueueResponse => {
+  const [pendingItems, setPendingItems] = useState<UploadItem[]>([]);
+
+  const { save } = useSaveData(undefined, {
+    postOptions: {
+      qsEncodeData: false,
```

Comment:
is this true by default for posts?

Reply:
Yes. the undefined key is getting converted into true.

```
const encodeData = options?.qsEncodeData == undefined || options?.qsEncodeData;
```